src/remote_connector_dao/GraphWidget2D.py

```diff
@@ -0,0 +1,68 @@
+import pyqtgraph as pg
+from remote_connector_dao.app_styles import (
+    colors,
+    graph_line_style,
+    axis_label_style,
+    graph_title_style,
+)
+
+LINE_COLORS = [
+    colors["red_primary"],
+    colors["blue_primary"],
+    colors["green_primary"],
+    colors["black"],
+]
+
+LINE_SYMBOLS = ["s", "o", "t", "+"]
+
+
+class GraphWidget2D(pg.PlotWidget):
+    def __init__(
+        self,
+        number_of_lines,
+        graph_lines_data,
+        background_color,
+        graph_title,
+        vertical_axis_label,
+        graph_line_labels_list,
+    ):
+        super().__init__()
+        self.graph_lines = []
+        self.graph_lines_data = graph_lines_data
+        self.number_of_lines = number_of_lines
+        self.graph_line_labels_list = graph_line_labels_list
+
+        self.setBackground(background_color)
+        self.setTitle(graph_title, **graph_title_style)
+        self.setLabel("left", vertical_axis_label, **axis_label_style)
+        self.addLegend(offset=(10, 10))
+        self.showGrid(x=True, y=True)
+        self.setMouseEnabled(x=False, y=True)
+
+        self._create_graph_lines()
+
+    def _create_graph_lines(self):
+        for i in range(self.number_of_lines):
```

Comment:
O que acontece aqui se `self.number_of_lines` for maior que a quantidade de items disponiveis em `LINE_COLORS` (no caso 4).

Me parece que na linha abaixo estamos tentando fazer um acesso inseguro que pode quebrar caso o valor passado para `number_of_lines` seja maior que nossa `LINE_COLORS`

Reply:
It is not safe to access, @cassios discussed with me and we will create a line object to hold all properties of a line, so when we attempt to create a line we must pass color to this line.

I am thinking that to create these lines under an interaction based on the user choice of many lines, I can maybe implement logic to my interaction over the `colors` list to be modular, it will repeat the colors in the worst case, but it is very similar to the `cycler` logic of Matplotlib.